### PR TITLE
update CI permissions

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
CI now needs to ability to check the existence of the db backups bucket.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -145,6 +147,13 @@ jobs:
           cd infra
           $terraform init
 
+          echo $IS_MASTER
```
Comment (shayne-fletcher) [inline azure-pipelines.yml:150]: Do we mean to keep this?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -145,6 +147,13 @@ jobs:
...
+          if [ "$IS_MASTER" = "False" ]; then
```
Comment (shayne-fletcher) [inline azure-pipelines.yml:151]: Is this condition correct?

File: azure-pipelines.yml (hunk 1 above)
Comment (garyverhaegen-da) [reply azure-pipelines.yml]: No, see below.

File: azure-pipelines.yml (hunk 2 above)
Comment (garyverhaegen-da) [reply azure-pipelines.yml:150]: I'm still in the process of figuring that one out. Azure Pipelines docs are not super easy to interpret (though they usually end up making sense _after_ I've figured out the answer by trial-and-error).

I believe it's correct now though.

**Comment by garyverhaegen-da** [comment]
I've already applied the permissions manually, as the passing test confirms.

Bit more context: since I've added a new bucket (think hard drive in the cloud) for DB backups, and the CI job has been designed to run with the  minimum set of permissions it needs, the CI job now fails on missing permissions for that new bucket.

It doesn't need to do anything else than check the bucket already exists, but even for that it needs specific permission.

Now, I could just merge the changes to the `permissions` folder. But the missing permission got merged to `master` and was only caught after failing there, so I thought I'd add a check to PRs by running `terraform plan`, which is explicitly designed to prepare a plan of what it would do and print it, but not actually do anything.

And then I kinda fell into a rabbit hole of YAML programming. I wanted to make the check in the same "step", because I did not want to have to duplicate the definition of the `$terraform` command. But then I had to discover that Azure expressions are not supported in `env` values but are supported in `variables`, which in turn is not a step attribute but a job attribute, and this is all making me a bit sad but at least now it's working.

**Comment by shayne-fletcher** [comment]
> I've already applied the permissions manually, as the passing test confirms.
> 
> Bit more context: since I've added a new bucket (think hard drive in the cloud) for DB backups, and the CI job has been designed to run with the minimum set of permissions it needs, the CI job now fails on missing permissions for that new bucket.
> 
> It doesn't need to do anything else than check the bucket already exists, but even for that it needs specific permission.
> 
> Now, I could just merge the changes to the `permissions` folder. But the missing permission got merged to `master` and was only caught after failing there, so I thought I'd add a check to PRs by running `terraform plan`, which is explicitly designed to prepare a plan of what it would do and print it, but not actually do anything.
> 
> And then I kinda fell into a rabbit hole of YAML programming. I wanted to make the check in the same "step", because I did not want to have to duplicate the definition of the `$terraform` command. But then I had to discover that Azure expressions are not supported in `env` values but are supported in `variables`, which in turn is not a step attribute but a job attribute, and this is all making me a bit sad but at least now it's working.

So the idea is to run the plan if not master right? In which case the condition looks reasonable but probably worthy of a comment.

**Comment by shayne-fletcher** [comment]
I feel for you here. Even in my limited experience I've come to know that  Azure makes simple things hard somehow 😃

File: azure-pipelines.yml (hunk 2 above)
Comment (hurryabit) [reply azure-pipelines.yml:150]: I guess @shayne-fletcher's question is why you wouldn't want to sanity check on master. At least that's my question. 😸

**Comment by garyverhaegen-da** [comment]
> So the idea is to run the plan if not master right? In which case the condition looks reasonable but probably worthy of a comment.

Good point, will add one.

File: azure-pipelines.yml (hunk 2 above)
Comment (garyverhaegen-da) [reply azure-pipelines.yml:157]: That is a great question. The simple answer is I thought we already did, but re-reading more carefully it doesn't look like we do. I've moved the check outside of the conditional.

**Pull request merged**